BinaryEncoding.md

```diff
@@ -249,18 +329,39 @@ The data section declares the initialized data that is loaded
 into the linear memory.
 
 | Field | Type | Description |
-| ----- |  ----- | ----- |
+| ----- | ---- | ----------- |
 | count | `varuint32` | count of data segments to follow |
 | entries | `data_segment*` | repeated data segments as described below |
 
 a `data_segment` is:
 
 | Field | Type | Description |
-| ----- |  ----- | ----- |
-| offset | `varuint32` | the offset in linear memory at which to store the data |
+| ----- | ---- | ----------- |
+| index | `varuint32` | the [linear memory index](Modules.md#linear-memory-index-space) (0 in the MVP) |
+| offset | `init_expr` | an `i32` initializer expression that computes the offset at which to place the data |
 | size | `varuint32` | size of `data` (in bytes) |
 | data | `bytes` | sequence of `size` bytes |
 
+### Element section
+
+ID: `elem`
```

Comment:
Nit: So far we have written out all section names. If we take "elem", then I'd suggest shortening "function" to "func", too. We probably should separate that Q, so I suggest `element` for this PR.